sort_plain_test.go

```diff
@@ -12,4 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package certsort_test
+package certsort
+
+import (
+	"testing"
+)
+
+var filePaths = []string{
+	"test_data/concat/ca.pem",
+	"test_data/concat/cert.pem",
+	"test_data/concat/key.pem",
+}
+
+// Note that these test are not testing the actual sorting algorithm, but rather the
+// the error or invalid input outcome, which is basically garbage in, garbage out.
+// Makes sure it does not panic or return unexpected results.
+
+func TestCertificateSort(t *testing.T) {
+	rawKeys, err := readFiles(filePaths)
+	if err != nil {
+		t.Fatalf("Error in readFiles: %s", err)
+	}
+
+	certs := NewRawCerts(string(rawKeys[0]))
+	certs = certs.Append(string(rawKeys[1]))
+	certs = certs.Append(string(rawKeys[2]))
+
+	sortedCerts, key, err := SortCertificates(certs, true)
+	if err != nil {
+		t.Fatalf("Error in SortCertificates: %s", err)
+	}
+	if len(sortedCerts) != 6013 {
+		t.Fatalf("Expected chain length to be 3, got: %d", len(sortedCerts))
+	}
+	if key == nil {
+		t.Fatal("Key was expected to be non-nil")
+	}
+}
+
+func TestInvalidCaOnly(t *testing.T) {
```

Comment:
this is the test that would have failed without the code change right? maybe comment this in the test?